src/Microsoft.ML.LightGbm/LightGbmArguments.cs

```diff
@@ -1,678 +1,225 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using System.Reflection;
-using System.Text;
 using Microsoft.ML;
 using Microsoft.ML.CommandLine;
 using Microsoft.ML.EntryPoints;
 using Microsoft.ML.Internal.Internallearn;
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Trainers.LightGbm;
 
-[assembly: LoadableClass(typeof(Options.TreeBooster), typeof(Options.TreeBooster.Options),
-    typeof(SignatureLightGBMBooster), Options.TreeBooster.FriendlyName, Options.TreeBooster.Name)]
-[assembly: LoadableClass(typeof(Options.DartBooster), typeof(Options.DartBooster.Options),
-    typeof(SignatureLightGBMBooster), Options.DartBooster.FriendlyName, Options.DartBooster.Name)]
-[assembly: LoadableClass(typeof(Options.GossBooster), typeof(Options.GossBooster.Options),
-    typeof(SignatureLightGBMBooster), Options.GossBooster.FriendlyName, Options.GossBooster.Name)]
+[assembly: LoadableClass(typeof(GradientBooster), typeof(GradientBooster.Options),
+    typeof(SignatureLightGBMBooster), GradientBooster.FriendlyName, GradientBooster.Name)]
+[assembly: LoadableClass(typeof(DartBooster), typeof(DartBooster.Options),
+    typeof(SignatureLightGBMBooster), DartBooster.FriendlyName, DartBooster.Name)]
+[assembly: LoadableClass(typeof(GossBooster), typeof(GossBooster.Options),
+    typeof(SignatureLightGBMBooster), GossBooster.FriendlyName, GossBooster.Name)]
 
-[assembly: EntryPointModule(typeof(Options.TreeBooster.Options))]
-[assembly: EntryPointModule(typeof(Options.DartBooster.Options))]
-[assembly: EntryPointModule(typeof(Options.GossBooster.Options))]
+[assembly: EntryPointModule(typeof(GradientBooster.Options))]
+[assembly: EntryPointModule(typeof(DartBooster.Options))]
+[assembly: EntryPointModule(typeof(GossBooster.Options))]
 
 namespace Microsoft.ML.Trainers.LightGbm
 {
     internal delegate void SignatureLightGBMBooster();
 
     [TlcModule.ComponentKind("BoosterParameterFunction")]
-    public interface ISupportBoosterParameterFactory : IComponentFactory<IBoosterParameter>
-    {
-    }
-
-    public interface IBoosterParameter
+    internal interface IBoosterParameterFactory : IComponentFactory<BoosterParameterBase>
     {
```

Comment:
Should this be a best friend? I don't know if command line framework can call it now.

Reply:
I can run the command line. 

---
In reply to: [265815086](https://github.com/dotnet/machinelearning/pull/2948#discussion_r265815086) [](ancestors = 265815086)